diff --git a/m4/python.m4 b/m4/python.m4
index 2d3b654b8..b44d72561 100644
--- a/m4/python.m4
+++ b/m4/python.m4
@@ -94,7 +94,7 @@ AC_DEFUN([AC_PYTHON_DEVEL],[
 	#
 	AC_MSG_CHECKING([for a version of Python >= '2.1.0'])
 	ac_supports_python_ver=`$PYTHON -c "import sys, string; \
-		ver = string.split(sys.version)[[0]]; \
+		ver = sys.version_info[:3]; \
 		print ver >= '2.1.0'"`
 	if test "$ac_supports_python_ver" != "True"; then
 		if test -z "$PYTHON_NOVERSIONCHECK"; then
@@ -121,7 +121,7 @@ to something else than an empty string.
 	if test -n "$1"; then
 		AC_MSG_CHECKING([for a version of Python $1])
 		ac_supports_python_ver=`$PYTHON -c "import sys, string; \
-			ver = string.split(sys.version)[[0]]; \
+			ver = sys.version_info[:3]; \
 			print ver >= '$1'"`
 		if test "$ac_supports_python_ver" = "True"; then
 	   	   AC_MSG_RESULT([yes])
